src/parser.c

```diff
@@ -313,24 +313,28 @@ static exprtree parse_number(parser_t parser) {
     if (parser->pos+1 < parser->ntokens) {
         switch (parser->tokens[parser->pos]) {
             case '0':
-                if (parser->tokens[parser->pos+1] == 'b') {
-                    // Enter if number is binary
-                    numbertype = BIN_TYPE;
-                    parser->pos += 2;
-                    break;
-                } else if (parser->tokens[parser->pos+1] == 'x'){
-                    // Enter is number is hex
-                    goto hex;
-                } else {
-                    // Enter if number is decimal
-                    break;
+                // check second character
+                switch(parser->tokens[parser->pos+1]) {
+                    case 'b': // 0b0101
+                        numbertype = BIN_TYPE;
+                        parser->pos += 2;
+                        break;
+                    case 'x': // 0xffff
+                        numbertype = HEX_TYPE;
+                        parser->pos += 2;
+                        break;
+                    default:  // number is decimal
+                        break;
                 }
-            case 'x':
-hex:
+                break;
+            case 'x': // xffff
                 numbertype = HEX_TYPE;
-                // If x was 1st character add 1 otherwise add 2 to parser->pos
-                parser->pos += (parser->tokens[parser->pos] == 'x') ? 1 : 2;
-
+                parser->pos += 1;
+                break;
+            case 'b': // b0101
+                numbertype = BIN_TYPE;
+                parser->pos += 1;
+                break;
```

Comment:
Maybe this could be refactored a little but i am not sure.
What are your thoughts on this? @alt-romes @demo-exe 
I know `goto`'s can be evil but in a small use case they can simplify things without reducing the code quality too much.
```
    int numbertype = DEC_TYPE;
    int increment_parser_pos = 1;
    if (parser->pos+1 < parser->ntokens) {
        switch (parser->tokens[parser->pos]) {
            case '0':
                // check second character
                switch(parser->tokens[parser->pos+1]) {
                    case 'b': // 0b0101
                        increment_parser_pos = 2;
                        goto binary;
                    case 'x': // 0xffff
                        increment_parser_pos = 2;
                        goto hex;
                    default:  // number is decimal
                        break;
                }
                break;
            case 'x': // xffff
hex:
                numbertype = HEX_TYPE;
                parser->pos += increment_parser_pos;
                break;
            case 'b': // b0101
binary:
                numbertype = BIN_TYPE;
                parser->pos += increment_parser_pos;
                break;
        }

    }

```

Reply:
So for me personally, not using gotos is "simpler" to read. But I'll let you guys decide.

Reply:
The current implementation doesnt look bad. I just thought that could reduce duplicate code in the future.
Maybe a function or macro would serve equally good.

Reply:
I don't think gotos will reduce code duplication, only thing that gets 'deduplicated' is the set of numbertype. This is not enough of a gain for a tree with 5 branches. Code with just 2 nested switches is much less spaghetti-like

Reply:
I agree that currently the gotos dont save anything really but in future i can definetly see some saving potential.
Maybe the solution is to move the entire switch case to a function and seperate detecting numerical system and parsing number.

Like when we validate if a number is really in the correct numerical system.
In the current implementation the number 0b1023 is translated to 2 but it acutally should throw an error.
This piece of code could be the proper spot for callung a validate function.

This is just an idea from my side and i wont reject the merge if you and @alt-romes want to.

Reply:
One place to implement such an error would be later, when consuming char by char. This one calls for an issue

Reply:
Yeah you are right, the validate function is best positioned a bit later.

Reply:
Agreed.

I think both alternatives look good (with and without gotos) -- but since the code is already without gotos and such a validation layer would be a potential future improvement, we can merge it as is and revisit it later.

The goal of this MR is the binary numbers without leading zero feature, which it clearly acomplishes.

Thanks!